proto/api/v2/vulnerability.proto

```diff
@@ -0,0 +1,135 @@
+syntax = "proto3";
+
+package v2;
+
+import "api/v2/vuln_state.proto";
+import "google/protobuf/timestamp.proto";
+
+option go_package = "./api/v2;v2";
+option java_package = "io.stackrox.proto.api.v2";
+
+enum VulnerabilityType {
+  UNKNOWN_VULNERABILITY = 0;
+  IMAGE_VULNERABILITY = 1;
+  K8S_VULNERABILITY = 2;
+  ISTIO_VULNERABILITY = 3;
+  NODE_VULNERABILITY = 4;
+  OPENSHIFT_VULNERABILITY = 5;
+  VIRTUAL_MACHINE_VULNERABILITY = 6;
+}
+
+message EmbeddedVulnerability {
```

Comment:
We need to be very careful here since we are introducing a new v2 API. I don't have any concrete concerns, but I also don't know much about the existing API objects. Is there anything we want to change for v2? Let's make sure many engineers see this to avoid any unforeseen pitfalls from copy&pasting the v1 protos.